src/compiler/utilities.ts

```diff
@@ -2414,6 +2414,27 @@ namespace ts {
         return output;
     }
 
+    export const stringify: (value: any) => string = JSON && JSON.stringify ? JSON.stringify : function stringify(value: any): string {

```

Comment:
But it doesn't have a note or assertion limiting to such usage and its in the general `utilities` file and is marked as a fallback for `JSON.stringify` (which correctly throws a `TypeError: Converting circular structure to JSON` in such a scenario).

You should probably at least leave a note about the limitation.

It's not necessarily a limitation you'd inherently expect.